Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/Settings.java

```diff
@@ -249,6 +260,32 @@ public void saveSettings(SettingsActivityView view, Context context, Set<String>
     }
   }
 
+  public static void manageGCAdapterScanThread()
+  {
+    boolean useAdapter = false;
+
+    for (int i = 0; i < 4; i++)
+    {
+      if (NativeLibrary
+              .GetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", Settings.SECTION_INI_CORE,
+                      SettingsFile.KEY_GCPAD_TYPE + i, InputOverlay.EMULATED_GAMECUBE_CONTROLLER)
+              .equals(InputOverlay.GAMECUBE_ADAPTER))
```

Comment:
This function should be written in C++. That way, you can check ConfigManager's m_SIDevice array instead of having to read from the INI file.

Reply:
Now that I look, it seems like there already is a C++ function that does this checking: https://github.com/dolphin-emu/dolphin/blob/bed2d66bedca7e9537d67d3e7a5506aee8717c8f/Source/Core/InputCommon/GCAdapter_Android.cpp#L378-L385

Reply:
Better?